provider/cluster_rosa_classic_resource.go

```diff
@@ -1370,6 +1375,26 @@ func (r *ClusterRosaClassicResource) upgradeClusterIfNeeded(ctx context.Context,
 		// letting it fail w/ an error. The user can then ack the agreement via
 		// some other method and re-run the apply.
 
+		clusterClient := r.clusterCollection.Cluster(state.ID.Value)
```

Comment:
I think those lines (1378-1397) can be removed to a different method 
Also, it's not clear what is the purpose of those lines in that section of the function